keps/sig-autoscaling/2556-circumvent-50-tag-asg-limit/README.md

```diff
@@ -0,0 +1,139 @@
+1. Cluster Autoscaler will then check the ASG tags for the EKS ManagedNodegroups tags. ([ASG tags code][])
+1. If Cluster Autoscaler finds the EKS ManagedNodegroups tags, it will call methods on the ManagedNodegroupCache to get labels, taints, and some other values.
+   1. If the cache has data for the given ManagedNodegroup, it will return that data.
+   1. If the data’s TTL has expired, the DescribeNodegroup API will be called and the data will be added to the cache and returned.
+   1. If there is no data cached for the ManagedNodegroup, the DescribeNodegroup API will be called and the data will be added to the cache and returned.
+1. Then Cluster Autoscaler will pull information from the ASG tags like it already does. If duplicate keys are found, the values from the ASG tags are kept.
+1. Cluster Autoscaler will then continue as usual.
+
+[allocatable resources code]: https://github.com/kubernetes/autoscaler/blob/99936f3f3971c91b30def913c1c807c42c4b87a7/cluster-autoscaler/cloudprovider/aws/aws_manager.go#L366
+[ASG tags code]: https://github.com/kubernetes/autoscaler/blob/99936f3f3971c91b30def913c1c807c42c4b87a7/cluster-autoscaler/cloudprovider/aws/aws_manager.go#L370
+
+## Testing
+
+- [TC1] Test creating a DescribeNodegroup request
+- [TC2] Test pulling values from cache when data exists for the ManagedNodegroups
+- [TC3] Test pulling values from cache and calling DescribeNodegroup API if no data exists
+- [TC4] Test missing permissions and falling back to just ASG tags
+- [TC5] Test duplicate label keys - selecting the value from ASG tags response
+
+## Risks
+
+By default, Cluster Autoscaler runs every 10 seconds. Our best practices documentation notes that this short interval can cause throttling because Cluster Autoscaler already makes AWS API calls during each loop. Our documentation recommends that customers increase the interval, so adding this API shouldn’t cause latency problems for customers. ([EKS Best Practices][]) Also, the API call will only happen for EKS ManagedNodegroups. If this increase in latency is too much for even one run of the loop, we will look into moving the API calls into parallel goroutines. 
+
+EKS also throttles describe API calls by default. To mitigate this issue, Cluster Autoscaler will keep a cache of DescribeNodegroup responses: ManagedNodegroupCache. Each cache bucket will have a TTL and, when the TTL expires, DescribeNodegroup will be called again. If there are errors during the call to DescribeNodegroup, Cluster Autoscaler will move on and just look at the ASG tags and existing default allocatable resource values.
+
+[EKS Best Practices]: https://aws.github.io/aws-eks-best-practices/cluster-autoscaling/cluster-autoscaling/#reducing-the-scan-interval
+
+## Other Solutions Considered
+
+Creating a configMap in the cluster that saves all of the node information for ManagedNodegroups was considered.
```

Comment:
I'm unclear, would this proposed solution have been done by the ManagedNodegroup control plane in AWS creating and updating this configMap in the cluster?

Reply:
Yes, that's correct.